### PR TITLE
Bluetooth: Audio: Fix bad docs for codec helper functions

### PR DESCRIPTION
**Pull request by Thalley**
Several of the codec helper functions used @retval incorrectly Several functions referenced @ccid_list incorrectly

### CODE REVIEW

File: include/zephyr/bluetooth/audio/audio.h [hunk 1]
```diff
@@ -845,9 +845,9 @@ int bt_audio_codec_cfg_freq_hz_to_freq(uint32_t freq_hz);
  * @param codec_cfg The codec configuration to extract data from.
  *
  * @retval A @ref bt_audio_codec_cfg_freq value
```
Comment (kartben) [inline include/zephyr/bluetooth/audio/audio.h:847]: Missed this one

File: include/zephyr/bluetooth/audio/audio.h [hunk 2]
```diff
@@ -890,9 +890,9 @@ int bt_audio_codec_cfg_frame_dur_us_to_frame_dur(uint32_t frame_dur_us);
  * @param codec_cfg The codec configuration to extract data from.
  *
  * @retval A @ref bt_audio_codec_cfg_frame_dur value
```
Comment (kartben) [inline include/zephyr/bluetooth/audio/audio.h:892]: Ditto

File: include/zephyr/bluetooth/audio/audio.h [hunk 3]
```diff
@@ -824,8 +824,8 @@ enum bt_audio_dir {
  *
  * @param freq The assigned numbers frequency to convert.
  *
- * @retval -EINVAL if arguments are invalid.
- * @retval The converted frequency value in Hz.
+ * @retval -EINVAL Arguments are invalid.
+ * @retval frequency The converted frequency value in Hz.
```
Comment (kartben) [inline include/zephyr/bluetooth/audio/audio.h:828]: You want this one listed first I think, i.e. the most likely return value first (applies throughout the PR)

Related, and since we don't have guidelines just yet and I very much value your thought on this topic and the fact we can use this PR as a way to settle on best practices for documenting return values, how about the "non-distinctive" value be described using `@return`? Basically, this: https://docs.rtems.org/branches/master/eng/coding-doxygen.html#function-declarations

**Review by kartben — CHANGES_REQUESTED** [review]

File: include/zephyr/bluetooth/audio/audio.h (hunk 3 above)
Comment (Thalley) [reply include/zephyr/bluetooth/audio/audio.h:827]: > how about the "non-distinctive" value be described using @return

So this is suggesting to use `@retval` for specific values, and `@return` for values that may range?

So in this example this would be something like 

```c
 * @retval -EINVAL Arguments are invalid.
 * 
 * @return The converted frequency value in Hz.
```

I'm not opposed to it, depending on how it would render in the built docs :) It does make sense to only use `@retval` for specific values. Speaking of, should we use `- @ref EINVAL`? or add something like a `@errno{EINVAL}`? 


I think it makes sense to propose an RFC to discuss this

Comment (kartben) [reply include/zephyr/bluetooth/audio/audio.h:827]: > Speaking of, should we use `- @ref EINVAL`? or add something like a `@errno{EINVAL}`?

Good question. Given that retval accepts ONE token representing the return value, the former likely won't work, and not sure about the latter either. That being said, I like the idea as making it easier to navigate to the list of available errnos can only be goodness.

+1 to address the next iteration (combining usage of `@return` and `@retval`) in a follow-up RFC

**Review by kartben — DISMISSED** [review]

File: include/zephyr/bluetooth/audio/audio.h [hunk 4]
```diff
@@ -777,8 +777,8 @@ int bt_audio_data_parse(const uint8_t ltv[], size_t size,
  *                  Any found data will be little endian.
  *
  * @retval Length The length of found @p data (may be 0).
```
Comment (kruithofa) [inline include/zephyr/bluetooth/audio/audio.h:779]: Maybe not part of this PR, but using a lowercase `l` is consistent with the parameter description in the other 
 fucntions
```suggestion
 * @retval length The length of found @p data (may be 0).
```

File: include/zephyr/bluetooth/audio/audio.h [hunk 5]
```diff
@@ -857,9 +857,9 @@ int bt_audio_codec_cfg_get_freq(const struct bt_audio_codec_cfg *codec_cfg);
  * @param codec_cfg The codec configuration to set data for.
  * @param freq      The assigned numbers frequency to set.
  *
- * @retval The data_len of @p codec_cfg on success
- * @retval -EINVAL if arguments are invalid
- * @retval -ENOMEM if the new value could not set or added due to memory
+ * @retval data_len The data_len of @p codec_cfg on success
```
Comment (kruithofa) [inline include/zephyr/bluetooth/audio/audio.h:860]: In some other functions where the return value is `len` the description says `length of ...`, maybe the same should be done for `data_len` and `meta_len`, here and in other functions
```suggestion
 * @retval data_len The data length of @p codec_cfg on success
```

**Review by kruithofa — COMMENTED** [review]
LGTM, just some nitpicks

File: include/zephyr/bluetooth/audio/audio.h (hunk 5 above)
Comment (Thalley) [reply include/zephyr/bluetooth/audio/audio.h:860]: Hmm, I get your point. `data_len` length here refer to the `data_len` field of `codec_cfg`. Will rewrite a bit

Comment (kartben) [reply include/zephyr/bluetooth/audio/audio.h:860]: +1, this too caught my eye fwiw :)

File: include/zephyr/bluetooth/audio/audio.h [hunk 6]
```diff
@@ -1083,9 +1083,9 @@ int bt_audio_codec_cfg_meta_get_val(const struct bt_audio_codec_cfg *codec_cfg,
  * @param data       Pointer to the data-pointer to set.
  * @param data_len   Length of @p data.
  *
- * @retval The meta_len of @p codec_cfg on success
- * @retval -EINVAL if arguments are invalid
- * @retval -ENOMEM if the new value could not set or added due to memory
+ * @retval meta_len The meta_len of @p codec_cfg on success
```
Comment (kruithofa) [inline include/zephyr/bluetooth/audio/audio.h:1086]: Shall we do the same as for the data_len field of codec_cfg?

```suggestion
 * @retval meta_len The @p codec_cfg.meta_len on success
```

File: include/zephyr/bluetooth/audio/audio.h [hunk 7]
```diff
@@ -1099,8 +1099,8 @@ int bt_audio_codec_cfg_meta_set_val(struct bt_audio_codec_cfg *codec_cfg,
  * @param codec_cfg  The codec data to set the value in.
  * @param type       The type id to unset.
  *
- * @retval The meta_len of @p codec_cfg on success
- * @retval -EINVAL if arguments are invalid
+ * @retval meta_len The meta_len of @p codec_cfg on success
```
Comment (kruithofa) [inline include/zephyr/bluetooth/audio/audio.h:1102]:
```suggestion
 * @retval meta_len The @p codec_cfg.meta_len on success
```

**Review by kruithofa — DISMISSED** [review]
One nitpick, but nothing blocking

File: include/zephyr/bluetooth/audio/audio.h (hunk 6 above)
Comment (Thalley) [reply include/zephyr/bluetooth/audio/audio.h:1086]: Good catch, will update

**Review by kruithofa — APPROVED** [review]

**Review by kartben — APPROVED** [review]
